packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -546,15 +544,6 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
   /// actions to filter changes, it must be called again (during the layout of
   /// the nearest descendant [RenderObjectWidget] of the gesture detector).
   void replaceSemanticsActions(Set<SemanticsAction> actions) {
-    assert(() {
-      if (!context.findRenderObject().owner.debugDoingLayout) {
-        throw new FlutterError(
-          'Unexpected call to replaceSemanticsActions() method of RawGestureDetectorState.\n'
-          'The replaceSemanticsActions() method can only be called during the layout phase.'
-        );
-      }
-      return true;
-    }());
```

Comment:
This seems scary; why is it ok to do this now?

Reply:
When I introduced this method I thought it only ever needs to be called during layout and to prevent missuse I guarded it with this assert (similarly to what you did with replaceGestureRecognizers above). However, it turns out that the SignleChildScrollView doesn't participate in the layout phase after scrolling because its layout didn't changed. Nevertheless, we still need to adjust what semantics actions are available after the scroll, which means we need to loosen when this method can be called.

Reply:
If it's called before layout, then won't this crash? The render object won't be ready yet.